pyresample/ewa/ewa.py

```diff
@@ -149,7 +149,9 @@ def fornav(cols, rows, area_def, data_in,
     """
     if isinstance(data_in, (tuple, list)):
         # we can only support one data type per call at this time
-        assert(in_arr.dtype == data_in[0].dtype for in_arr in data_in[1:])
+        for in_arr in data_in[1:]:
+           if in_arr.dtype != data_in[0].dtype:
+               raise ValueError("All input arrays must be the same dtype")
```

Comment:
```suggestion
        for in_arr in data_in[1:]:
            if in_arr.dtype != data_in[0].dtype:
                raise ValueError("All input arrays must be the same dtype")
```

That's what I get for using github's web editor.